### PR TITLE
Add E2E tests

### PR DESCRIPTION
**Pull request by Raalsky**

### CODE REVIEW

File: kedro_neptune/__init__.py [hunk 1]
```diff
@@ -210,13 +207,13 @@ def _describe(self) -> Dict[str, Any]:
     def _save(self, data: bytes) -> None:
         load_path = get_filepath_str(self._get_load_path(), self._protocol)
```
Comment (pkasprzyk) [inline kedro_neptune/__init__.py]: to further avoid confusion, you can change `load_path` -> `save_path` :)

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    assert run.exists(f'{dataset_namespace}/type')
+    assert run.exists(f'{dataset_namespace}/version')
+
+
+def check_node_metadata(run: neptune.Run, node_namespace: str, inputs: List, outputs: Optional[List] = None):
+    # pylint: disable=eval-used
+    assert run.exists(node_namespace)
+    assert run.exists(f'{node_namespace}/execution_time')
+    assert run.exists(f'{node_namespace}/inputs')
+    assert sorted(eval(run[f'{node_namespace}/inputs'].fetch())) == inputs
```
Comment (pkasprzyk) [inline tests/conftest.py]: I don't really like this `eval`.. is this only because `assign()` does `str()`?

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            custom_run_id=custom_run_id,
+            capture_stderr=False,
+            capture_stdout=False,
+            capture_hardware_metrics=False,
+            source_files=[]
+        )
+
+
+def check_dataset_metadata(run: neptune.Run, dataset_namespace: str):
+    assert run.exists(f'{dataset_namespace}/filepath')
```
Comment (pkasprzyk) [inline tests/conftest.py]: `exists()` is a rather poor check :( couldn't you verify expected values?

File: tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        run_params: Dict[str, Any],
+        session_params: Dict[str, Any]
+):
+    if 'NEPTUNE_CUSTOM_RUN_ID' in os.environ:
+        del os.environ['NEPTUNE_CUSTOM_RUN_ID']
+
+    with KedroSession.create(project, **session_params) as session:
+        session.run(**run_params)
+
+        run_id = session.store["session_id"]
```
Comment (pkasprzyk) [inline tests/conftest.py:43]: `session_id` is globally unique, right?

Comment (Raalsky) [reply tests/conftest.py:43]: It's timestamp of pipeline startup.

File: kedro_neptune/__init__.py (hunk 1 above)
Comment (Raalsky) [reply kedro_neptune/__init__.py]: I've renamed it to just `path`.

File: tests/conftest.py (hunk 2 above)
Comment (Raalsky) [reply tests/conftest.py]: Neptune uploads arrays as String :<. Now it's changed to `ast.literal_eval` which is much safer than simple eval.

File: tests/conftest.py (hunk 3 above)
Comment (Raalsky) [reply tests/conftest.py]: Sure, changed to fetching and dict compare

**Review by pkasprzyk — APPROVED** [review]

**Pull request merged**